rudder-core/src/main/scala/com/normation/rudder/domain/queries/CmdbQuery.scala

```diff
@@ -436,74 +436,194 @@ case object EditorComparator extends CriterionType {
   override def toLDAP(value:String) = Full(value)
 }
 
-case class JsonComparator(key:String, splitter:String = "", numericvalue:Boolean = false) extends TStringComparator with Loggable {
-  override val comparators = HasKey +: BaseComparators.comparators
+/*
+ * This comparator is used to look for a specific key=value in a json value, where key is known
+ * before hand. Typically, that comparator is used when you serialized several field in one json,
+ * and you need to lookup one of these field's value.
+ * Ex:
+ * LDAP attritude and value:
+ *    my_serialized_data: {"key1": "value1", "key2": "value2" }
+ *
+ * JsonFixedKeyComparator("my_serialized_data", "key2", false) => will search for value in "value2"
+ *
+ * Used for "process" attribute
+ */
+case class JsonFixedKeyComparator(ldapAttr:String, jsonKey: String, numericvalue: Boolean) extends TStringComparator with Loggable {
+  override val comparators = BaseComparators.comparators
+
+  def format(attribute:String, value:String) = {
+    val v = if (numericvalue) value else s""""$value""""
```

Comment:
is numeric value treating boolean and  maybe other json type (array ? object ?)

Reply:
it's a very special use case, dedicated to process, and we don't have anything else than numeric. But I could call it "quoteValue", it would make more sense, you are right